lib/plugins.py

```diff
@@ -483,9 +483,24 @@ def run(self):
 
 
 def hook(func):
-    hook_names.add(func.__name__)
+    func._is_ec_plugin_hook = True
     return func
 
+def _get_func_if_hook(plugin, attr_name) -> Optional[Callable]:
+    cls = plugin.__class__
+    cls_attr = getattr(cls, attr_name, None)
+    if isinstance(cls_attr, property):
+        # The queried attr_name corresponds to an instance property. We avoid
+        # calling getattr on instance properties below (since that has side
+        # effects). Just return False as that's the answer
+        return
+    # Ok, attr_name wasn't a property. So it's safe to call getattr on it to
+    # figure out if it's a function, and if so, if it has the special
+    # _is_ec_plugin_hook attribute we may have tagged it with in @hook
+    func = getattr(plugin, attr_name, None)
+    if callable(func) and getattr(func, '_is_ec_plugin_hook', False):
+        return func
```

Comment:
could be just:
```
def _get_func_if_hook(plugin, attr_name) -> Optional[Callable]:
    cls = plugin.__class__
    cls_attr = getattr(cls, attr_name, None)
    if getattr(cls_attr, '_is_ec_plugin_hook', False):
        return getattr(plugin, attr_name, None)
```

no?

Reply:
I have no idea lol.. let me check

What I did works (I tested it).. but let me check if what you propose works too..

Reply:
Both work.  It's actually a class property though on the function instance so what you propose is "more correct" in that it doesn't grab the stupid bound method which presumably is slower.  I'll fix it.

Reply:
Meh. I'll leave it.  We're grabbing the bound method anyway later for possible return.  I feel the code is more readable to just check on the constraint right before you return the thing, rather than check the constraint on the class-level thing.  *shrug*

Reply:
OK .. you were right. fixed.